R/args.R

```diff
@@ -147,15 +149,38 @@ SampleArgs <- R6::R6Class(
       self$thin <- thin
       self$max_depth <- max_depth
       self$metric <- metric
-      # self$metric_file <- character()
+      self$inv_metric <- inv_metric
+      if(!is.null(inv_metric)) {
+        if(!is.null(metric_file)) {
+          stop("Only one of inv_metric and metric_file can be non-NULL")
+        }
+
+        # wrap inv_metric in list if not in one
+        if(!is.list(inv_metric)) {
+          inv_metric = list(inv_metric)
+        }
+
+        # write all inv_metrics to disk
+        inv_metric_paths <- c()
+        for(metric_i in 1:length(inv_metric)) {
+          inv_metric_path <-
+            file.path(
+              cmdstan_tempdir(),
+              paste0("inv_metric-", metric_i, ".json")
+            )
+          write_stan_json(list(inv_metric = inv_metric[[metric_i]]), inv_metric_path)
+          inv_metric_paths <- c(inv_metric_paths, inv_metric_path)
+        }
+
+        self$metric_file <- inv_metric_paths
+      } else {
+        self$metric_file <- sapply(metric_file, repair_path)
+        self$metric_file <- sapply(self$metric_file, absolute_path)
```

Comment:
I think `absolute_path` calls `repair_path` internally, so you just need `absolute_path`. Di you do this because I use them both back to back in this order somewhere else? If so maybe I did that for a reason but I'd have to take a look.

Reply:
Also should I sapply here or should we just wrap the utilities with vectorize?

Reply:
Also absolute_path does call repair path, good call, fixed

Reply:
We should probably vectorize `absolute_path` but not a problem if you want to leave it with `sapply` for now.